packages/leap/lib/src/tiled_object_factory.dart

```diff
@@ -0,0 +1,10 @@
+import 'package:flame/components.dart';
+import 'package:flame_tiled/flame_tiled.dart';
+
+/// Interface for creating components from Tiled Objects.
+mixin TiledObjectFactory<T extends Component> {
+  /// Returns a Flame [Component] from a Tiled Object. Implementers
+  /// can use the [TiledObject.x], [TiledObject.y], [TiledObject.properties],
+  /// etc. to customize the component.
+  T createComponent(TiledObject tiledObject);
```

Comment:
One minor suggestion here, would be for this method to also receive the game instance.

This could be useful in case for some reason, the creation of the component depend in information available in the game itself.

One clear useful instance is if we want to create the component using the images/assets instance of the game itself.

Reply:
Updated this to receive the `leapMap` so you can access the game off of that, and the `layer` the object was in

Reply:
Looks great!